modules/fargate/fargate.tf

```diff
@@ -1,19 +1,19 @@
 resource "aws_iam_role" "eks_fargate_pod" {
-  count              = local.create_eks && var.create_fargate_pod_execution_role ? 1 : 0
+  count              = var.create_eks && var.create_fargate_pod_execution_role ? 1 : 0
   name_prefix        = format("%s-fargate", var.cluster_name)
   assume_role_policy = data.aws_iam_policy_document.eks_fargate_pod_assume_role[0].json
   tags               = var.tags
   path               = var.iam_path
 }
 
 resource "aws_iam_role_policy_attachment" "eks_fargate_pod" {
-  count      = local.create_eks && var.create_fargate_pod_execution_role ? 1 : 0
+  count      = var.create_eks && var.create_fargate_pod_execution_role ? 1 : 0
   policy_arn = "${var.iam_policy_arn_prefix}/AmazonEKSFargatePodExecutionRolePolicy"
   role       = aws_iam_role.eks_fargate_pod[0].name
 }
 
 resource "aws_eks_fargate_profile" "this" {
-  for_each               = local.create_eks ? local.fargate_profiles_expanded : {}
+  for_each               = var.create_eks ? local.fargate_profiles_expanded : {}
```

Comment:
This `for_each` will fail if `local.fargate_profiles_expanded` contains not-yet-known values (as you've just replaced for `local.create_eks`). I am not sure what is the best solution for this here.

Reply:
`var.fargate_profiles` and `local.fargate_profiles_expanded` are maps and to work with `for_each`, their keys must be known during plan. If keys are known, I think the Terraform will wait correctly for `aws_eks_fargate_profile` creation if there is a not yet known values inside `local.fargate_profiles_expanded`.

So `length(var.fargate_profiles)` will be known because keys must be known.